relay/nakama/save.go

```diff
@@ -0,0 +1,190 @@
+		return logErrorFailedPrecondition(
+			logger,
+			eris.New("data cannot be empty"),
+		)
+	}
+
+	// get the persona storage object.
+	persona, err := loadPersonaTagStorageObj(ctx, nk)
+	if err != nil {
+		if eris.Is(eris.Cause(err), ErrPersonaTagStorageObjNotFound) {
+			return "", eris.Wrap(err, "cannot save game: user has not yet claimed a persona tag")
+		}
+		return logErrorFailedPrecondition(logger, eris.Wrap(err, "failed to get persona for save"))
+	}
+	// do not allow saving if they do not yet have an accepted tag.
+	if persona.Status != personaTagStatusAccepted {
+		return logErrorFailedPrecondition(
+			logger,
+			eris.Errorf("persona tag %q is not yet verified: cannot save", persona.PersonaTag),
+		)
+	}
+
+	// check if the user is allowlisted. NOTE: checkVerified will return nil in two cases:
+	// case 1: if the allowlist is disabled (via ENABLE_ALLOWLIST env var).
+	// case 2: the user is actually allowlisted.
+	var verified bool
+	err = checkVerified(ctx, nk, userID)
+	if err != nil {
+		// as long as the error isn't that they're not allowlisted, return the error.
+		// we ignore the ErrNotAllowlisted, which will keep verified == false.
+		if !eris.Is(eris.Cause(err), ErrNotAllowlisted) {
+			return logErrorFailedPrecondition(logger, eris.Wrap(err, "could not read verification table"))
+		}
+	} else {
+		// when err == nil, that means checkVerified passed, or that there is no allowlist enabled.
+		// so we just set verified to true.
+		verified = true
+	}
+
+	save := Save{
+		Data:     msg.Data,
+		Persona:  persona.PersonaTag,
+		Verified: verified,
+	}
+	saveBz, err := json.Marshal(save)
+	if err != nil {
+		return logErrorFailedPrecondition(logger, eris.Wrap(err, "failed to marshal save file"))
+	}
+
+	err = writeSave(ctx, userID, string(saveBz), nk)
+	if err != nil {
+		return logErrorFailedPrecondition(
+			logger,
+			eris.Wrap(err, "failed to write game save to storage"),
+		)
+	}
+
+	response, err := json.Marshal(SaveGameResponse{Success: true})
+	if err != nil {
+		return logErrorFailedPrecondition(logger, eris.Wrap(err, "failed to marshal response"))
+	}
+
+	return string(response), nil
+}
+
+func writeSave(ctx context.Context, userID string, save string, nk runtime.NakamaModule) error {
+	write := &runtime.StorageWrite{
+		Collection:      gameSaveCollection,
+		Key:             userID,
+		UserID:          userID,
+		Value:           save,
+		Version:         "",
+		PermissionRead:  runtime.STORAGE_PERMISSION_OWNER_READ,
+		PermissionWrite: runtime.STORAGE_PERMISSION_OWNER_WRITE,
+	}
+	_, err := nk.StorageWrite(ctx, []*runtime.StorageWrite{write})
+	return err
+}
+
+func initSaveFileQuery(_ runtime.Logger, initializer runtime.Initializer) error {
+	err := initializer.RegisterRpc(
+		"get-save",
+		handleQueryGameSave,
+	)
+	if err != nil {
+		return eris.Wrap(err, "")
+	}
+	return nil
+}
+
+func handleQueryGameSave(ctx context.Context, logger runtime.Logger, _ *sql.DB, nk runtime.NakamaModule, _ string,
```

Comment:
nit, "handleGetSaveGame" to match the endpoint string

Reply:
done